Parser/Node.cs

```diff
@@ -20,18 +22,30 @@ public record NBlock (NDeclarations Declarations, NCompoundStmt Body) : Node {
 }
 
 // The declarations section precedes the body of every block
-public record NDeclarations (NVarDecl[] Vars, NFnDecl[] Funcs) : Node {
+public record NDeclarations (NConstDecl[] Consts, NVarDecl[] Vars, NFnDecl[] Funcs) : Node {
    public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
 }
 
+// Abstract class to declare function parameters
+public abstract record NParamDecl (Token Name, NType Type) : Node {
+   public override string ToString () => $"{Type} {Name}";
+}
+
 // Declares a variable (with a type)
-public record NVarDecl (Token Name, NType Type) : Node {
+public record NVarDecl (Token Name, NType Type) : NParamDecl (Name, Type) {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+// Declares a constant
+public record NConstDecl (Token Name, NType Type) : NParamDecl (Name, Type) {

```

Comment:
Not clear how an NConstDecl derives from NParamDecl (which is defined as an abstract class to declare function parameters)...